modules/account-lib/src/coin/stx/transactionBuilderFactory.ts

```diff
@@ -0,0 +1,71 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { BufferReader, deserializeTransaction, PayloadType } from '@stacks/transactions';
+import { BaseTransactionBuilderFactory } from '../baseCoin';
+import { ParseTransactionError, InvalidTransactionError, NotImplementedError } from '../baseCoin/errors';
+import { TransferBuilder } from './transferBuilder';
+import { TransactionBuilder } from './transactionBuilder';
+import { Transaction } from './transaction';
+
+export class TransactionBuilderFactory extends BaseTransactionBuilderFactory {
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  /** @inheritdoc */
+  from(raw: string): TransactionBuilder {
+    this.validateRawTransaction(raw);
+    try {
+      const tx = this.parseTransaction(raw);
+      switch (tx.stxTransaction.payload.payloadType) {
+        case PayloadType.TokenTransfer:
+          return this.getTransferBuilder(tx);
+        // TODO: add case of wallet
+        default:
+          throw new InvalidTransactionError('Invalid transaction');
+      }
+    } catch (e) {
+      console.error(e);
+      throw new ParseTransactionError('There was an error parsing the raw transaction');
+    }
+  }
+
+  private parseTransaction(rawTransaction: string): Transaction {
+    const tx = new Transaction(this._coinConfig);
```

Comment:
can we validate the raw transaction here as well? Or is it assumed it will always be validated by this point?

Reply:
Yes, it will always be validated by this point